gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/BasePage.java

```diff
@@ -310,10 +335,76 @@ protected final void defaultRoleChecksForInstructorOnlyPage()
 			case STUDENT:
 				throw new RestartResponseException(StudentPage.class);
 			default:
-				if(businessService.isUserAbleToEditAssessments()) {
+				if(businessService.isUserAbleToEditAssessments(getCurrentSiteId())) {
 					break;
 				}
 				throw new RestartResponseException(GradebookPage.class);
 		}
 	}
+
+	protected String getCurrentSiteId() {
+		try {
+			return this.toolManager.getCurrentPlacement().getContext();
+		} catch (final Exception e) {
+			return null;
+		}
+	}
+
+	protected String getCurrentGradebookUid() {
+		String gradebookUid = getCurrentSiteId();
+		Placement placement = toolManager.getCurrentPlacement();
+		Properties props = placement.getPlacementConfig();
+		if (props.getProperty("gb-group") != null) {
+			gradebookUid = props.getProperty("gb-group");
+		}
+
+		return gradebookUid;
+	}
+
+	public User getCurrentUser() {
+		return this.userDirectoryService.getCurrentUser();
+	}
+
+	/**
+	 * Get the user's custom GbUiSettings from PreferencesService
+	 *
+	 * @return String
+	 */
+	public String getUserGbPreference(final String prefName) {
+		final String siteId = getCurrentSiteId();
+		final String currentUserId = getCurrentUser().getId();
+		Preferences userPrefs = preferencesService.getPreferences(currentUserId);
+		ResourceProperties rp = userPrefs.getProperties(GB_PREF_KEY + siteId);
+		return rp.getProperty(prefName);
+	}
+
+	/**
+	 * Set the user's custom GbUiSettings in PreferencesService
+	 *
+	 * @return
+	 */
+	public void setUserGbPreference(final String prefName, final String prefValue) {
+		final String siteId = getCurrentSiteId();
+		final String currentUserId = getCurrentUser().getId();
+		PreferencesEdit prefsEdit = null;
+		try {
+			prefsEdit = preferencesService.edit(currentUserId);
+		}
+		catch (IdUnusedException e) {
+			try {
+				prefsEdit = preferencesService.add(currentUserId);
+			} catch (PermissionException e1) {
+				log.warn("setUserGbPreference PermissionException attempting to add prefs for user {}, prefName={}", currentUserId, prefName);
+			} catch (IdUsedException e1) {
+				log.warn("setUserGbPreference IdUsedException attempting to add prefs for user {}, prefName={}", currentUserId, prefName);
+			}
+		} catch (PermissionException e) {
+			log.warn("setUserGbPreference PermissionException attempting to edit prefs for user {}, prefName={}", currentUserId, prefName);
+		} catch (InUseException e) {
+			log.warn("setUserGbPreference InUseException attempting to edit prefs for user {}, prefName={}", currentUserId, prefName);
+		}
+		ResourcePropertiesEdit props = prefsEdit.getPropertiesEdit(GB_PREF_KEY + siteId);
+		props.addProperty(prefName, prefValue);
+		preferencesService.commit(prefsEdit);
```

Comment:
check elsewhere in sakai for a better way to make edits to a users preferences

Reply:
This was moved from the GradebookNgBusinessService class, as we tried to avoid to get current context or user on that class. But we didn't add or modify that part.